PWGCF/TwoParticleCorrelations/Tasks/neutronProtonCorrZdc.cxx

```diff
@@ -0,0 +1,247 @@
+    histos.add("diffZPASignal", "diffZPASignal", kTH1F, {axisZDiffSignal});
+    histos.add("diffZPCSignal", "diffZPCSignal", kTH1F, {axisZDiffSignal});
+    histos.add("diffZNSignal", "diffZNSignal", kTH1F, {axisZDiffSignal});
+    histos.add("diffZPSignal", "diffZPSignal", kTH1F, {axisZDiffSignal});
+
+    histos.add("CentralityPercentile", "CentralityPercentile", kTH1F, {cfgAxisCent});
+
+    histos.add("CentvsZNASignal", "FT0CvsZNASignal", kTH2F, {cfgAxisCent, axisZNASignal});
+    histos.add("CentvsZNCSignal", "FT0CvsZNCSignal", kTH2F, {cfgAxisCent, axisZNCSignal});
+    histos.add("CentvsZPASignal", "FT0CvsZPASignal", kTH2F, {cfgAxisCent, axisZPASignal});
+    histos.add("CentvsZPCSignal", "FT0CvsZPCSignal", kTH2F, {cfgAxisCent, axisZPCSignal});
+    histos.add("CentvsZNSignal", "FT0CvsZNSignal", kTH2F, {cfgAxisCent, axisZNSignal});
+    histos.add("CentvsZPSignal", "FT0CvsZPSignal", kTH2F, {cfgAxisCent, axisZPSignal});
+  }
+
+  void processRun3(soa::Filtered<soa::Join<aod::Collisions, aod::EvSels, CentralitiesRun3>>::iterator const& collision, BCsRun3 const&, aod::Zdcs const&)
+  {
+    if (!collision.sel8()) {
+      histos.fill(HIST("eventCounter"), 0.25);
+      return;
+    }
+    const auto& foundBC = collision.foundBC_as<BCsRun3>();
+    if (foundBC.has_zdc()) {
+      const auto& zdcread = foundBC.zdc();
+      const auto cent = collision.centFT0C();
+
+      histos.fill(HIST("eventCounter"), 1.25);
+      histos.fill(HIST("ZNASignal"), zdcread.energyCommonZNA());
+      histos.fill(HIST("ZNCSignal"), zdcread.energyCommonZNC());
+      histos.fill(HIST("ZPASignal"), zdcread.energyCommonZPA());
+      histos.fill(HIST("ZPCSignal"), zdcread.energyCommonZPC());
+      histos.fill(HIST("ZNASector0Signal"), (zdcread.energySectorZNA())[0]);
+      histos.fill(HIST("ZNASector1Signal"), (zdcread.energySectorZNA())[1]);
+      histos.fill(HIST("ZNASector2Signal"), (zdcread.energySectorZNA())[2]);
+      histos.fill(HIST("ZNASector3Signal"), (zdcread.energySectorZNA())[3]);
+      histos.fill(HIST("ZNCSector0Signal"), (zdcread.energySectorZNC())[0]);
+      histos.fill(HIST("ZNCSector1Signal"), (zdcread.energySectorZNC())[1]);
+      histos.fill(HIST("ZNCSector2Signal"), (zdcread.energySectorZNC())[2]);
+      histos.fill(HIST("ZNCSector3Signal"), (zdcread.energySectorZNC())[3]);
+      histos.fill(HIST("ZPASector0Signal"), (zdcread.energySectorZPA())[0]);
+      histos.fill(HIST("ZPASector1Signal"), (zdcread.energySectorZPA())[1]);
+      histos.fill(HIST("ZPASector2Signal"), (zdcread.energySectorZPA())[2]);
+      histos.fill(HIST("ZPASector3Signal"), (zdcread.energySectorZPA())[3]);
+      histos.fill(HIST("ZPCSector0Signal"), (zdcread.energySectorZPC())[0]);
+      histos.fill(HIST("ZPCSector1Signal"), (zdcread.energySectorZPC())[1]);
+      histos.fill(HIST("ZPCSector2Signal"), (zdcread.energySectorZPC())[2]);
+      histos.fill(HIST("ZPCSector3Signal"), (zdcread.energySectorZPC())[3]);
```

Comment:
I would suggest having a look at the https://github.com/AliceO2Group/O2Physics/blob/master/Tutorials/src/histogramRegistry.cxx tutorial, concretely the `fillHistos` function within the `RealisticExample` task to get inspiration on ways of addressing this